### PR TITLE
Fix: REST API issue with blocks registered via PHP.

### PR DESCRIPTION
**Pull request by AbhijitNage123**
### Description

- Resolved display conditions conflicting with dynamic blocks from various plugins.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### How has this been tested?

- Drag and drop Gravity Form block and its preview is visible in the editor or not.
- Drag and drop WP Recipe Maker plugins blocks and its preview is visible in the editor or not.

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: classes/class-uagb-loader.php [hunk 1]
```diff
@@ -130,6 +130,59 @@ public function load_plugin() {
...
+			add_filter( 'rest_pre_dispatch', array( $this, 'rest_pre_dispatch' ), 10, 3 );
+		}
+
+		/**
+		 * Fix REST API issue with blocks registered via PHP register_block_type.
+		 *
+		 * @param mixed  $result  Response to replace the requested version with.
+		 * @param object $server  Server instance.
+		 * @param object $request Request used to generate the response.
+		 *
```
Comment (RajkiranBagal) [inline classes/class-uagb-loader.php:142]: Add since tag

File: classes/class-uagb-loader.php [hunk 2]
```diff
@@ -130,6 +130,59 @@ public function load_plugin() {
...
+		 * @return array Returns updated results.
+		 */
+		public function rest_pre_dispatch( $result, $server, $request ) {
+
+			if ( strpos( $request->get_route(), '/wp/v2/block-renderer' ) !== false ) {
+
+				if ( isset( $request['attributes'] ) && isset( $request['attributes']['UAGHideDesktop'] ) ) {
```
Comment (RajkiranBagal) [inline classes/class-uagb-loader.php:149]: Common out the isset($request['attributes]) condition

File: classes/class-uagb-loader.php [hunk 3]
```diff
@@ -130,6 +130,59 @@ public function load_plugin() {
...
+
+					$attributes = $request['attributes'];
+					unset( $attributes['UAGHideDesktop'] );
```
Comment (RajkiranBagal) [inline classes/class-uagb-loader.php:152]: Convert to Switch case & common out the 1st & last statement of each condition too.

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: classes/class-uagb-loader.php (hunk 1 above)
Comment (AbhijitNage123) [reply classes/class-uagb-loader.php:144]: Updated

File: classes/class-uagb-loader.php (hunk 2 above)
Comment (AbhijitNage123) [reply classes/class-uagb-loader.php]: Updated

File: classes/class-uagb-loader.php (hunk 3 above)
Comment (AbhijitNage123) [reply classes/class-uagb-loader.php]: Updated

File: classes/class-uagb-loader.php [hunk 4]
```diff
@@ -130,6 +130,60 @@ public function load_plugin() {
...
+		 * @param object $server  Server instance.
+		 * @param object $request Request used to generate the response.
+		 *
+		 * @return array Returns updated results.
+		 */
+		public function rest_pre_dispatch( $result, $server, $request ) {
+
+			if ( strpos( $request->get_route(), '/wp/v2/block-renderer' ) !== false ) {
+
+				if ( isset( $request['attributes'] ) ) {
```
Comment (RajkiranBagal) [inline classes/class-uagb-loader.php:151]: Combine these 2 if statements

File: classes/class-uagb-loader.php [hunk 5]
```diff
@@ -130,6 +130,60 @@ public function load_plugin() {
...
+					$attributes = $request['attributes'];
+
+					switch ( $attributes ) {
+
+						case in_array( $attributes['UAGHideDesktop'], $attributes, true ):
```
Comment (RajkiranBagal) [inline classes/class-uagb-loader.php:156]: Need to change this switch. Will discuss on call,

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: classes/class-uagb-loader.php (hunk 5 above)
Comment (AbhijitNage123) [reply classes/class-uagb-loader.php]: Updated as discussed

File: classes/class-uagb-loader.php (hunk 4 above)
Comment (AbhijitNage123) [reply classes/class-uagb-loader.php]: Updated

**Review by RajkiranBagal — APPROVED** [review]

**Comment by akshaylakh** [comment]
working fine

**Review by sandeshjangam — APPROVED** [review]

**Pull request merged**